pkg/alertmanager/multitenant.go

```diff
@@ -199,6 +200,10 @@ func (cfg *MultitenantAlertmanagerConfig) Validate(storageCfg alertstore.Config)
 		}
 	}
 
+	if cfg.ShardingEnabled && cfg.ShardingRing.ZoneAwarenessEnabled && cfg.ShardingRing.InstanceZone == "" {
+		return errZoneAwarenessEnabledWithoutZoneInfo
+	}
```

Comment:
**suggestion:** I believe this if statements could be moved into the above if statement which would allow you to remove the `cfg.ShardingEnabled` condition.

```suggestion
		if cfg.ShardingRing.ZoneAwarenessEnabled && cfg.ShardingRing.InstanceZone == "" {
			return errZoneAwarenessEnabledWithoutZoneInfo
		}
	}
```